pkg/controller/networkpolicy/status_controller.go

```diff
@@ -274,12 +274,28 @@ func (c *StatusController) syncHandler(key string) error {
 		status := &crdv1alpha1.NetworkPolicyStatus{
 			Phase:              crdv1alpha1.NetworkPolicyPending,
 			ObservedGeneration: internalNP.Generation,
+			Conditions:         GenerateNetworkPolicyCondition(""),
 		}
 		if internalNP.SourceRef.Type == controlplane.AntreaNetworkPolicy {
 			return c.npControlInterface.UpdateAntreaNetworkPolicyStatus(internalNP.SourceRef.Namespace, internalNP.SourceRef.Name, status)
 		}
 		return c.npControlInterface.UpdateAntreaClusterNetworkPolicyStatus(internalNP.SourceRef.Name, status)
 	}
+
+	// It means the NetworkPolicy has been processed, and marked as unrealizable. It will enter unrealizable phase
+	// instead of being further realized. Meanwhile Antrea-agents will not report statuses.
+	if internalNP.RealizableMessage != "" {
+		status := &crdv1alpha1.NetworkPolicyStatus{
+			Phase:              crdv1alpha1.NetworkPolicyRealized,
```

Comment:
Showing a policy as "Realized" when it's not realizable may be confusing. Perhaps "Pending" is better here, then user would notice it and look for more detailed status.